src/v/cluster/log_eviction_stm.h

```diff
@@ -83,6 +84,11 @@ class log_eviction_stm final
     /// The actual start offset of the log with the delta factored in
     model::offset effective_start_offset() const;
 
+    /// Ensure followers have processed up until the most recent known version
+    /// of the batch representing the start offset
+    ss::future<result<model::offset, cluster::errc>>
```

Comment:
i think we shouldn't limit ourselves to `cluster::errc` simple `result<model::offset>` will handle errors coming from all the layers then we can differentiate them using error category.